src/migrator/QueryFactory.ts

```diff
@@ -60,33 +60,35 @@ export class QueryFactory {
 
   public async exec(consul: IMigrationClient): Promise<void> {
     if (this._isJson) {
-      const result = await consul.get<string>(this.key)
-      const parsed = typeof result === 'string' ? JSON.parse(result) : result
+      const result = await consul.get<string>(this.key);
+      const parsed = typeof result === 'string' ? JSON.parse(result) : result;
       if (!parsed) {
-        throw new Error('Current value is not a json object')
+        throw new Error('Current value is not a json object');
       }
+      const helper = new JSONHelper(result);
       if (this._jsonpath) {
-        jp.apply(
-          parsed,
-          this._jsonpath,
-          this._jsonpathFunc ?? (() => this._value)
-        )
-        return consul.set(this.key, JSON.stringify(parsed, null, 2))
+        if(jp.query(parsed, this._jsonpath).length) {
+          jp.apply(
+            parsed,
+            this._jsonpath,
+            this._jsonpathFunc ?? (() => this._value)
+          )
+        } else {
+          helper.insertObjectAtJPath(this._jsonpath, this._value);
+        }
+        return consul.set(this.key, JSON.stringify(parsed, null, 2));
```

Comment:
` } else {
          helper.insertObjectAtJPath(this._jsonpath, this._value);
        }
        return consul.set(this.key, JSON.stringify(parsed, null, 2));`

should be:
 } else {
          helper.insertObjectAtJPath(this._jsonpath, this._value);
          return consul.set(this.key, helper.toString(true));
        }
        return consul.set(this.key, JSON.stringify(parsed, null, 2));

Reply:
added.